packages/cinterop/src/commonMain/kotlin/io/realm/kotlin/internal/interop/sync/SyncError.kt

```diff
@@ -27,7 +27,8 @@ data class SyncError constructor(
     val recoveryFilePath: String?,
     val isFatal: Boolean,
     val isUnrecognizedByClient: Boolean,
-    val isClientResetRequested: Boolean
+    val isClientResetRequested: Boolean,
+    val compensatingWrites: Array<CoreCompensatingWriteInfo>
```

Comment:
Are we guaranteed that this is only contain information when we should throw a `CompensatingWriteException` ?

Reply:
I think so, we are only surfacing compensating writes when we have a `CompensatingWriteException`.